### PR TITLE
Frontend cleanup

### PR DESCRIPTION
**Pull request by sgfost**
- Adds Github actions workflow that checks TS/Vue code for formatting/linter errors, runs when push/pr has changes in `frontend/`
- Does initial re-formatting
- Fixes linter warnings/errors (largely removing unused vars/imports, no real semantic changes)

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -38,17 +41,22 @@
     "yup": "^0.28.2"
   },
   "devDependencies": {
+    "@rushstack/eslint-patch": "^1.2.0",
```
Comment (alee) [inline frontend/package.json:44]: should add context: https://rushstack.io/ and https://github.com/microsoft/rushstack

**Review by alee — COMMENTED** [review]
just a few minor notes, great work as always! not blockers, we can create issues for anything worth considering

File: .github/workflows/check-frontend-code.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./frontend
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up node
+      uses: actions/setup-node@v3
+      with:
+        node-version: 16
```
[the remark this answers is not in the document]
Comment (sgfost) [reply .github/workflows/check-frontend-code.yml:23]: good catch, had just copied over from PoM on node 16

File: frontend/package.json (hunk 1 above)
Comment (sgfost) [reply frontend/package.json:44]: also good catch, I'm 99% sure this isn't actually being used

File: .git-blame-ignore-revs [hunk 3]
```diff
@@ -1,3 +1,6 @@
 # .git-blame-ignore-revs
 # run upgraded black==23.1.0
 90d176f65ec2212a3c15d80c2dbbeb04f1c77c97
+
+# format frontend code with prettier 02-02-2023
```
[the remark this answers is not in the document]
Comment (sgfost) [reply .git-blame-ignore-revs:5]: makes sense to me, date is probably unnecessary just need some quick context and anything else can be gleaned from the commit

**Pull request merged**